falcon/typing.py

```diff
@@ -40,6 +40,16 @@
 # Error serializers
 ErrorSerializer = Callable[['Request', 'Response', BaseException], Any]
 
+JSONSerializable = Union[
+    Dict[str, 'JSONSerializable'],
+    List['JSONSerializable'],
```

Comment:
It's actually sequence here I think. Tuple serialises to list.

But maybe we can add tuple instead of any sequence

Reply:
I'm not sure `Sequence` is correct here, but you can suggest what the best annotation for JSON serializable data is, I don't think it matters _that_ much for SSE events.

Re `Sequence`, it doesn't seem that `json` supports arbitrary sequences...
```python
import collections.abc
import json

class MyRange(collections.abc.Sequence):
    def __init__(self, n):
        self.n = n
    def __len__(self):
        return self.n
    def __getitem__(self, key):
        key = int(key)
        if not 0 <= key < self.n:
            raise IndexError('Segfault huh?')
        return key

if __name__ == '__main__':
    mr = MyRange(7)
    print(f'{mr[5]=}')
    items = tuple(mr)
    print(items)
    print(json.dumps(mr))
```
:arrow_down: 
`TypeError: Object of type MyRange is not JSON serializable`

Reply:
Ok. The current type seems fine, just add `Tuple['JSONSerializable',...]` to the union and it should be fine